src/recommendpolicy/recommendPolicy.go

```diff
@@ -143,22 +160,40 @@ func RecommendPolicyMain() {
 			DeployNsName = append(DeployNsName, *deploy)
 		}
 
-		for _, ns := range nsNotFilter {
+		for _, ns := range NsNotFilter {
 			if d.Namespace != ns {
-				generateHardenPolicy(d.Name, d.Namespace, d.Spec.Template.Labels)
+				policy = generateHardenPolicy(d.Name, d.Namespace, d.Spec.Template.Labels)
 			}
 		}
 	}
+	policies = append(policies, policy...)
+	for _, r := range replicaSets.Items {
+		for _, ns := range NsNotFilter {
+			if r.Namespace != ns && len(r.ObjectMeta.OwnerReferences) == 0 {
+				policy = generateHardenPolicy(r.Name, r.Namespace, r.Spec.Template.Labels)
```

Comment:
Assume there are multiple namespaces with statefulset. When this for loop ends it will contain the harden `policy` only for the last namespace. All other namespaces hardening policies will be lost/overwritten. Am I missing something here?

Reply:
thanks for looking into it, updated the pr, please check.